Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4385,19 +4347,17 @@ CMtaVersion CGame::CalculateMinClientRequirement()
             m_strPrevMinClientKickRequirement = strKickMin;
 
             // Do kicking
-            uint uiNumIncompatiblePlayers = 0;
-            for (std::list<CPlayer*>::const_iterator iter = g_pGame->GetPlayerManager()->IterBegin(); iter != g_pGame->GetPlayerManager()->IterEnd(); iter++)
-            {
-                CPlayer* pPlayer = *iter;
-                if (strKickMin > pPlayer->GetPlayerVersion())
-                {
-                    CStaticFunctionDefinitions::RedirectPlayer(pPlayer, "", 0, NULL);
-                    uiNumIncompatiblePlayers++;
-                }
-            }
+            const auto& players = g_pGame->GetPlayerManager()->GetAllPlayers();
+            const auto  playersRedirectedCount = std::count_if(players.begin(), players.end(), [&strKickMin](CPlayer* pPlayer) {
+                if (strKickMin <= pPlayer->GetPlayerVersion())
+                    return false;
+
+                CStaticFunctionDefinitions::RedirectPlayer(pPlayer, "", 0, NULL);
+                return true;
+            });
 
-            if (uiNumIncompatiblePlayers > 0)
-                CLogger::LogPrintf(SString("Forced %d player(s) to reconnect so they can update to %s\n", uiNumIncompatiblePlayers, *strKickMin));
+            if (playersRedirectedCount > 0)
+                CLogger::LogPrintf(SString("Forced %d player(s) to reconnect so they can update to %s\n", playersRedirectedCount, *strKickMin));
```

Comment:
> `%d player(s)`

I assume `playersRedirectedCount` is a `size_t` (should be`unsigned long` on Win32). Then you probably should change that to `%u player(s)`?
Also: Please don't use **auto** if the type isn't obvious.

Reply:
I should add a cast to `unsigned int` instead.